lib/matplotlib/offsetbox.py

```diff
@@ -1506,6 +1506,9 @@ def on_release(self, event):
             self.finalize_offset()
             self.got_artist = False
             if self._use_blit:
+                self.ref_artist.draw(
```

Comment:
You probably want to restore the background as well here or you will double-draw the legend (set it to have a colored and fraction alpha background to see the effect).

Reply:
Ah yeah that's right, added the restoration!